gcovr/utils.py

```diff
@@ -180,7 +180,12 @@ def __init__(self, regex, path_context=None):
 
 class Filter(object):
     def __init__(self, pattern):
-        self.pattern = re.compile(pattern)
+        cwd = os.getcwd()
+        # Guessing if file system is case insensitive.
+        # The working directory is not the root and accessible in upper and lower case.
+        is_fs_case_insensitive = ( cwd != os.path.sep ) and os.path.exists(cwd.upper()) and os.path.exists(cwd.lower())
```

Comment:
Linter doesn't like extra spaces.
```suggestion
        is_fs_case_insensitive = (cwd != os.path.sep) and os.path.exists(cwd.upper()) and os.path.exists(cwd.lower())
```